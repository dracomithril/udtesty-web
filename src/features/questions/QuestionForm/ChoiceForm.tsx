import React from 'react';
import styled from 'styled-components';
import { Checkbox } from 'antd';
import { CheckboxChangeEvent } from 'antd/lib/checkbox';
import Choice from './Choice';

export const CHOICE_CORRECT_CHECKBOX = 'choice-correct-checkbox';
export const CHOICE_TEST_ID = 'choice_container';

const ChoiceHeader = styled.header`
  display: flex;
  justify-content: space-between;
  flex: 1;
`;

const translations = {
  correct: 'poprawna',
};

const ChoiceForm = ({
  id,
  onSelectChange,
  correct,
  name,
  'data-testid': dataTestId,
  children,
  className,
}:
{
  correct: boolean,
  name: string,
  id: number,
  onSelectChange: (event: CheckboxChangeEvent) => void,
  children?: React.ReactNode,
  'data-testid'?: string,
  className?: string,
}) => (
  <Choice
    selected={correct}
    odd={id % 2 === 0}
    data-testid={dataTestId}
    className={className}
  >
    <ChoiceHeader>
      <span>{`${name}.`}</span>
      <label htmlFor={`correct.${name}`}>
        {translations.correct}
        <Checkbox
          id={`correct.${name}`}
          data-testid={CHOICE_CORRECT_CHECKBOX}
          name={name}
          checked={correct}
          onChange={onSelectChange}
        />
      </label>
    </ChoiceHeader>
    {children}
  </Choice>
);

ChoiceForm.defaultProps = {
  className: '',
  children: undefined,
  'data-testid': CHOICE_TEST_ID,
};

export default ChoiceForm;
