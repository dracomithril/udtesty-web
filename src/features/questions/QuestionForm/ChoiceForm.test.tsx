import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ChoiceForm, { CHOICE_CORRECT_CHECKBOX } from './ChoiceForm';

describe('ChoiceForm', () => {
  it('should react on correct answer change', () => {
    const name = 'A';
    const onAnswerSelectedChange = jest.fn();
    const { baseElement, getByTestId } = render(<ChoiceForm
      correct
      name={name}
      id={3}
      onSelectChange={onAnswerSelectedChange}
    />);
    const correctAnswer = getByTestId(CHOICE_CORRECT_CHECKBOX);
    expect(correctAnswer).toBeChecked();
    fireEvent.click(correctAnswer);
    expect(onAnswerSelectedChange).toBeCalled();
    expect(baseElement).toMatchSnapshot();
  });
});
