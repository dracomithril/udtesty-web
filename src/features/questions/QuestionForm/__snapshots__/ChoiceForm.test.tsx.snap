// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoiceForm should react on correct answer change 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa nVCzP"
      data-testid="choice_container"
    >
      <header
        class="sc-bwzfXH eYXKZd"
      >
        <span>
          A.
        </span>
        <label
          for="correct.A"
        >
          poprawna
          <label
            class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
          >
            <span
              class="ant-checkbox ant-checkbox-checked"
            >
              <input
                checked=""
                class="ant-checkbox-input"
                data-testid="choice-correct-checkbox"
                id="correct.A"
                name="A"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
        </label>
      </header>
    </div>
  </div>
</body>
`;
